pkg/acsengine/defaults.go

```diff
@@ -705,98 +705,98 @@ func setHostedMasterProfileDefaults(a *api.Properties) {
 
-	provided := certsAlreadyPresent(a.CertificateProfile, a.MasterProfile.Count)
+	provided := certsAlreadyPresent(p.CertificateProfile, p.MasterProfile.Count)
 
 	if areAllTrue(provided) {
 		return false, nil
 	}
 
-	masterExtraFQDNs := append(formatAzureProdFQDNs(a.MasterProfile.DNSPrefix), a.MasterProfile.SubjectAltNames...)
-	firstMasterIP := net.ParseIP(a.MasterProfile.FirstConsecutiveStaticIP).To4()
+	masterExtraFQDNs := append(formatAzureProdFQDNs(p.MasterProfile.DNSPrefix), p.MasterProfile.SubjectAltNames...)
+	firstMasterIP := net.ParseIP(p.MasterProfile.FirstConsecutiveStaticIP).To4()
 
 	if firstMasterIP == nil {
-		return false, errors.Errorf("MasterProfile.FirstConsecutiveStaticIP '%s' is an invalid IP address", a.MasterProfile.FirstConsecutiveStaticIP)
+		return false, errors.Errorf("MasterProfile.FirstConsecutiveStaticIP '%s' is an invalid IP address", p.MasterProfile.FirstConsecutiveStaticIP)
 	}
 
 	ips := []net.IP{firstMasterIP}
-	// Add the Internal Loadbalancer IP which is always at at a known offset from the firstMasterIP
+	// Add the Internal Loadbalancer IP which is always at at p known offset from the firstMasterIP
 	ips = append(ips, net.IP{firstMasterIP[0], firstMasterIP[1], firstMasterIP[2], firstMasterIP[3] + byte(DefaultInternalLbStaticIPOffset)})
 	// Include the Internal load balancer as well
 
-	if a.MasterProfile.IsVirtualMachineScaleSets() {
+	if p.MasterProfile.IsVirtualMachineScaleSets() {
 		// Include the Internal load balancer as well
-		for i := 1; i < a.MasterProfile.Count; i++ {
-			offset := i * a.MasterProfile.IPAddressCount
+		for i := 1; i < p.MasterProfile.Count; i++ {
+			offset := i * p.MasterProfile.IPAddressCount
 			ip := net.IP{firstMasterIP[0], firstMasterIP[1], firstMasterIP[2], firstMasterIP[3] + byte(offset)}
 			ips = append(ips, ip)
 		}
 	} else {
-		for i := 1; i < a.MasterProfile.Count; i++ {
+		for i := 1; i < p.MasterProfile.Count; i++ {
 			ip := net.IP{firstMasterIP[0], firstMasterIP[1], firstMasterIP[2], firstMasterIP[3] + byte(i)}
 			ips = append(ips, ip)
 		}
 	}
-	if a.CertificateProfile == nil {
-		a.CertificateProfile = &api.CertificateProfile{}
+	if p.CertificateProfile == nil {
+		p.CertificateProfile = &api.CertificateProfile{}
 	}
 
-	// use the specified Certificate Authority pair, or generate a new pair
-	var caPair *PkiKeyCertPair
+	// use the specified Certificate Authority pair, or generate p new pair
+	var caPair *helpers.PkiKeyCertPair
 	if provided["ca"] {
-		caPair = &PkiKeyCertPair{CertificatePem: a.CertificateProfile.CaCertificate, PrivateKeyPem: a.CertificateProfile.CaPrivateKey}
+		caPair = &helpers.PkiKeyCertPair{CertificatePem: p.CertificateProfile.CaCertificate, PrivateKeyPem: p.CertificateProfile.CaPrivateKey}
 	} else {
-		caCertificate, caPrivateKey, err := createCertificate("ca", nil, nil, false, false, nil, nil, nil)
+		var err error
+		caPair, err = helpers.CreatePkiKeyCertPair("ca")
 		if err != nil {
-			return false, err
+			return false, nil
```

Comment:
Shouldn't we continue to return the `err` here?

Reply:
Thanks for this.